tracer/src/Datadog.Trace/AppSec/Transports/Sender.cs

```diff
@@ -41,11 +41,14 @@ internal async Task Send(IEnumerable<IEvent> events)
             };
             var request = _apiRequestFactory.Create(_uri);
             request.AddHeader(AppSecHeaderKey, AppSecHeaderValue);
-            var response = await request.PostAsJsonAsync(batch, _serializer);
-            if (response.StatusCode != 200 && response.StatusCode != 202)
+            using (var response = await request.PostAsJsonAsync(batch, _serializer))
             {
-                var responseText = await response.ReadAsStringAsync();
-                Log.Warning($"AppSec event not correctly sent to backend {response.StatusCode} with response {responseText}");
+                if (response.StatusCode != 200 && response.StatusCode != 202)
+                {
+                    var statusCode = response.StatusCode;
+                    var responseText = await response.ReadAsStringAsync();
+                    Log.Warning($"AppSec event not correctly sent to backend {statusCode} with response {responseText}, request content was: {await request.RequestContent()}");
```

Comment:
nit: should use structured logging
```suggestion
                    Log.Warning("AppSec event not correctly sent to backend {StatusCode} with response {ResponseText}, request content was: {Content}", statusCode, responseText, await request.RequestContent());
```

Reply:
Thanks for the feedback! So I removed the implementation, and now all 3 impl. can log the payload, I just log directly in the method if the response status code is in error.